### PR TITLE
fix(service_integration): improve webhooks checks

### LINKED ISSUE #366: Can't create Datadog integration for existing PostgreSQL service
I'm not able to create datadog integration for existing PostgreSQL service. In the kubernetes logs i see message that preconditions are not met 
INFO controllers.ServiceIntegration preconditions are not met, requeue {"kind": "serviceintegration", "name": "aiven-operator/datadog-kotest", "annotations": {}}. 

Using the Aiven API, the same integration is created without any problems

### PR DESCRIPTION
**Pull request by byashimov**
- Adds `ServiceIntegration` kind `SourceProjectName` and `DestinationProjectName` fields
- Adds `ServiceIntegration` fields `MaxLength` validation
- Adds `ServiceIntegration` validation: multiple user configs cannot be set
- Fixes `ServiceIntegration`, should not require `destinationServiceName` or `sourceEndpointID` field
- Fixes `ServiceIntegration`, add missing `external_aws_cloudwatch_metrics` type config serialization
- Updates `ServiceIntegration` integration type list
- Drops ginkgo
- Setups webhooks in tests
- Fixed race condition in the test suite on `Destroy()` call

Resolves #366

### CODE REVIEW

File: charts/aiven-operator-crds/templates/aiven.io_serviceintegrations.yaml [hunk 1]
```diff
@@ -247,17 +247,26 @@ spec:
...
                   rule: self == oldSelf
               destinationServiceName:
                 description: Destination service for the integration (if any)
+                maxLength: 64
                 type: string
                 x-kubernetes-validations:
                 - message: Value is immutable
                   rule: self == oldSelf
-              external_aws_cloudwatch_metrics:
+              externalAWSCloudwatchMetrics:
```
Comment (Serpentiel) [inline charts/aiven-operator-crds/templates/aiven.io_serviceintegrations.yaml:269]: why did it change?

File: api/v1alpha1/serviceintegration_types.go [hunk 2]
```diff
@@ -25,26 +28,40 @@ type ServiceIntegrationSpec struct {
 	Project string `json:"project"`
 
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
-	// +kubebuilder:validation:Enum=datadog;kafka_logs;kafka_connect;metrics;dashboard;rsyslog;read_replica;schema_registry_proxy;jolokia;internal_connectivity;external_google_cloud_logging;datasource;clickhouse_postgresql;clickhouse_kafka;logs;external_aws_cloudwatch_metrics
-	// Type of the service integration
+	// +kubebuilder:validation:Enum=alertmanager,autoscaler,caching,cassandra_cross_service_cluster,clickhouse_kafka,clickhouse_postgresql,dashboard,datadog,datasource,external_aws_cloudwatch_logs,external_aws_cloudwatch_metrics,external_elasticsearch_logs,external_google_cloud_logging,external_opensearch_logs,flink,flink_external_kafka,internal_connectivity,jolokia,kafka_connect,kafka_logs,kafka_mirrormaker,logs,m3aggregator,m3coordinator,metrics,opensearch_cross_cluster_replication,opensearch_cross_cluster_search,prometheus,read_replica,rsyslog,schema_registry_proxy,stresstester,thanosquery,thanosstore,vmalert
+	// Type of the service integration accepted by Aiven API. Some values may not be supported by the operator
```
Comment (byashimov) [inline api/v1alpha1/serviceintegration_types.go:32]: Please have a look at this: it is really hard to say which types can be supported without user configuration. I think it is better put the whole list and leave a warning here.

File: api/v1alpha1/serviceintegration_types.go [hunk 3]
```diff
@@ -103,8 +120,33 @@ type ServiceIntegration struct {
...
+
+	// Checks if it is the only configuration set
+	for k, v := range configs {
+		if k != thisType && !reflect.ValueOf(v).IsNil() {
+			return nil, fmt.Errorf("got additional configuration for integration type %q", k)
+		}
+	}
+
+	return configs[thisType], nil
 }
```
Comment (byashimov) [inline api/v1alpha1/serviceintegration_types.go:150]: Unfortunately, I can't say for sure which type requires a configuration. E.g. `clickhouse_postgresql` comes with the only configuration field `Databases`, which can be [empty](https://github.com/aiven/aiven-operator/blob/3ff3d75f3d83eb17bc403c43a065359e81094edc/api/v1alpha1/userconfig/integration/clickhouse_postgresql/clickhouse_postgresql.go#L23-L23). Therefore, it is possible to create `clickhouse_postgresql` service integration without paticular configuration.

File: api/v1alpha1/serviceintegration_types.go [hunk 4]
```diff
@@ -70,7 +87,7 @@ type ServiceIntegrationSpec struct {
 	LogsUserConfig *logsuserconfig.LogsUserConfig `json:"logs,omitempty"`
 
 	// External AWS CloudWatch Metrics integration Logs configuration values
-	ExternalAWSCloudwatchMetricsUserConfig *externalawscloudwatchmetricsuserconfig.ExternalAwsCloudwatchMetricsUserConfig `json:"external_aws_cloudwatch_metrics,omitempty"`
+	ExternalAWSCloudwatchMetricsUserConfig *externalawscloudwatchmetricsuserconfig.ExternalAwsCloudwatchMetricsUserConfig `json:"externalAWSCloudwatchMetrics,omitempty"`
```
Comment (byashimov) [inline api/v1alpha1/serviceintegration_types.go:90]: This type never sent its [configuration](https://github.com/aiven/aiven-operator/blob/main/controllers/serviceintegration_controller.go#L170). So it doesn't break things.

**Review by ivan-savciuc — COMMENTED** [review]
Do not know if it is related but I get: 
```
=== CONT  TestServiceIntegrationKafkaLogs
    suite_test.go:142: panicked: runtime error: index out of range [2] with length 2
    suite_test.go:143: stacktrace: 
        goroutine 915 [running]:
        runtime/debug.Stack()
                /usr/lib/golang/src/runtime/debug/stack.go:24 +0x65
        github.com/aiven/aiven-operator/tests.recoverPanic(0xc000729a00)
                /home/ivan.savciuc/aiven-operator/tests/suite_test.go:143 +0x6d
        panic({0x183ebe0, 0xc000baf0e0})
                /usr/lib/golang/src/runtime/panic.go:884 +0x212
        github.com/aiven/aiven-operator/tests.(*session).Destroy(0xc000fb4370)
                /home/ivan.savciuc/aiven-operator/tests/suite.go:150 +0x157
        github.com/aiven/aiven-operator/tests.TestServiceIntegrationKafkaLogs(0xc000729a00)
                /home/ivan.savciuc/aiven-operator/tests/serviceintegration_test.go:254 +0xbaa
        testing.tRunner(0xc000729a00, 0x1a35768)
                /usr/lib/golang/src/testing/testing.go:1446 +0x10b
        created by testing.(*T).Run
                /usr/lib/golang/src/testing/testing.go:1493 +0x35f
--- FAIL: TestServiceIntegrationKafkaLogs (237.75s)
``` on this branch

**Comment by byashimov** [comment]
[tests](https://github.com/aiven/aiven-operator/actions/runs/4754822225/jobs/8448466111).

**Review by ivan-savciuc — APPROVED** [review]
LGTM

**Pull request merged**